Sources/swift-build/main.swift

```diff
@@ -48,6 +48,9 @@ do {
         let builddir = Path.join(getenv("SWIFT_BUILD_PATH") ?? Path.join(rootd, ".build"), configuration.dirname)
 
         for pkg in dependencies {
+            let manifest = try Manifest(path: "\(pkg.path)/Package.swift", baseURL: pkg.path)
+            let dependencyUrls = manifest.package.dependencies.map { $0.url.basename }
+            let dependencies = dependencies.filter { dependencyUrls.contains($0.url.basename) }
```

Comment:
`basename` is not enough, so this patch must change. The test should also verify that the thing you are working around by using basename is not a problem. Presumably this is due to relative file URLs.